app/models/queue_item.rb

```diff
@@ -23,18 +31,12 @@ def category_name
 
   private
 
-  def set_position
-    queue = self.user ? self.user.queue_items : [] # FIXME: just to pass the uniqueness shoulda matcher tests!!!
-    max_position = queue.empty? ? 0 : queue.count
-    self.position = max_position + 1
+  def review
+    @review ||= Review.find_by(creator: user, video: video)
   end
 
-  def reassign_positions
-    remaining_queue = QueueItem.all.select do |queue_item|
-      queue_item.user == self.user and queue_item.position > self.position
-    end
-    remaining_queue.each do |queue_item|
-      queue_item.update(position: queue_item.position - 1)
-    end
+  def set_position
+    queue = QueueItem.where(user: user)
+    self.position ||= queue.count + 1
```

Comment:
`self` must be here because you are setting the position. If it was not there position would be local variable. `self.position` calls for setter method. `user` above is a getter method provided by AR because it is a column in your queue_items table, not a local variable.